browser/brave_content_browser_client_browsertest.cc

```diff
@@ -740,3 +745,92 @@ IN_PROC_BROWSER_TEST_F(BraveContentBrowserClientTest, CheckExpectedExtensions) {
   // Brave Extension background page should be disabled by default.
   EXPECT_FALSE(extensions::BackgroundInfo::HasBackgroundPage(brave_extension));
 }
+
+#if BUILDFLAG(IS_WIN)
+
+class WinBraveContentBrowserClientTest
+    : public BraveContentBrowserClientTest,
+      public ::testing::WithParamInterface<
+          base::test::ScopedOSInfoOverride::Type> {
+ public:
+  WinBraveContentBrowserClientTest() = default;
+  ~WinBraveContentBrowserClientTest() override = default;
+
+  bool GetShouldDoLearning(Browser* browser) {
+    bool value = false;
+    browser->tab_strip_model()
+        ->GetActiveWebContents()
+        ->GetShouldDoLearningForTesting(value);
+    return value;
+  }
+
+  bool IsWin11() const {
+    return GetParam() == base::test::ScopedOSInfoOverride::Type::kWin11Home;
+  }
+
+ private:
+  base::test::ScopedOSInfoOverride win_version_{GetParam()};
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    ,
+    WinBraveContentBrowserClientTest,
+    ::testing::Values(base::test::ScopedOSInfoOverride::Type::kWin11Home,
+                      base::test::ScopedOSInfoOverride::Type::kWin10Home));
+
+IN_PROC_BROWSER_TEST_P(WinBraveContentBrowserClientTest, PRE_WindowsRecall) {
+  EXPECT_EQ(windows_recall::IsWindowsRecallAvailable(), IsWin11());
+
+  if (IsWin11()) {
```

Comment:
the result of the test should not be dependent on the version of windows this test runs on. You should mock the version to different values so you can test it correctly across a range of versions

Reply:
I see now https://github.com/brave/brave-core/pull/29251/files#r2132336920